dev/devicelab/lib/tasks/hot_mode_tests.dart

```diff
@@ -26,74 +28,76 @@ TaskFunction createHotModeTest({String deviceIdOverride, Map<String, String> env
-                process.stdin.writeln('q');
-              }
-            }
-            print('stdout: $line');
-          }, onDone: () {
-            stdoutDone.complete();
-          });
-          process.stderr
-              .transform<String>(utf8.decoder)
-              .transform<String>(const LineSplitter())
-              .listen((String line) {
-            print('stderr: $line');
-          }, onDone: () {
-            stderrDone.complete();
-          });
+      await inDirectory<void>(_editedFlutterGalleryDir, () async {
+        twoReloadsData = await captureReloadData(options, environment, benchmarkFile, (String line, Process process) {
+          if (!line.contains('Reloaded ')) {
+            return;
+          }
+          if (hotReloadCount == 0) {
+            // Update a file for 2-3 library invalidation.
+            final File appDartSource = file(path.join(
+              _editedFlutterGalleryDir.path, 'lib/gallery/app.dart',
+            ));
+            appDartSource.writeAsStringSync(
+              appDartSource.readAsStringSync().replaceFirst(
+                "'Flutter Gallery'", "'Updated Flutter Gallery'",
+              ));
+            process.stdin.writeln('r');
+            hotReloadCount += 1;
+          } else {
+            process.stdin.writeln('q');
+          }
+        });
 
-          await Future.wait<void>(
-              <Future<void>>[stdoutDone.future, stderrDone.future]);
-          await process.exitCode;
+        threeReloadsData = await captureReloadData(options, environment, benchmarkFile, (String line, Process process) {
+          if (!line.contains('Reloaded ')) {
+            return;
+          }
+          if (hotReloadCount == 1) {
+            // Update a file for 2-3 library invalidation.
+            final File appDartSource = file(path.join(
+              _editedFlutterGalleryDir.path, 'lib/demo/calculator/home.dart',
+            ));
+            appDartSource.writeAsStringSync(
+              appDartSource.readAsStringSync().replaceFirst(kSourceLine, kReplacementLine)
+            );
+            process.stdin.writeln('r');
+            hotReloadCount += 1;
+          } else {
+            process.stdin.writeln('q');
+          }
+        });
 
-          twoReloadsData = json.decode(benchmarkFile.readAsStringSync()) as Map<String, dynamic>;
-        }
-        benchmarkFile.deleteSync();
+        fourReloadsData = await captureReloadData(options, environment, benchmarkFile, (String line, Process process) {
+          if (!line.contains('Reloaded ')) {
+            return;
+          }
+          if (hotReloadCount == 2) {
+            // Trigger a framework invalidation (370 libraries) without modifying the source
+            final File flutterFrameworkSource = file(path.join(
+              flutterDirectory.path, 'packages/flutter/lib/src/widgets/framework.dart',
+            ));
+            flutterFrameworkSource.setLastModifiedSync(DateTime.now());
```

Comment:
Could this be flaky?

Reply:
I don't think it will in this case, but we would see that show up with large variations in the benchmark data